File: 2017/Python/day_1.py
def parse_input():
    splitter = ""
    with open("Inputs/day_1_input.txt") as input_file:
        for line in input_file:
            splitter = list(line)

            # Remove newline from the splitter
            splitter.pop()

    return splitter


def day_1():
    n_arr = parse_input()
    sumr = 0

    for i in range(len(n_arr)):
        num = int(n_arr[i])
        next_num = int(n_arr[0])

        if i != len(n_arr) - 1:
            next_num = int(n_arr[i + 1])

        if num == next_num:
            sumr += num
    print("The answer to Part 1 is : " + str(sumr))


def day_2():
    n_arr = parse_input()
    sumr = 0
    halfway = len(n_arr) / 2

    for i in range(len(n_arr)):
        num = int(n_arr[i])
        next_pos = int(i + halfway)

        if next_pos >= len(n_arr):
            next_pos = next_pos - len(n_arr)

        next_num = int(n_arr[next_pos])

        if num == next_num:
            sumr += num

    print("The answer to Part 2 is : " + str(sumr))


if __name__ == "__main__":
    day_1()
    day_2()

File: 2017/Rust/aoc_2017/src/day_1.rs
use std::fs;

pub fn day_1_main() {
    //Ingest data
    let file_name = "inputs/day_1_input.txt";
    let contents = fs::read_to_string(file_name).expect("Failed to read File");

    //Split by each line
    let mut parts: Vec<&str> = contents.split("").collect();

    //Get rid of the crap
    parts.remove(0);
    parts.remove(parts.len() - 1);
    parts.remove(parts.len() - 1);

    println!("The answer the Part 1 is: {}", part_1(parts.clone()));
    println!("The answer the Part 2 is: {}", part_2(parts.clone()));
}

fn part_1(parts: Vec<&str>) -> i32 {
    let mut sumr = 0;

    for i in 0..parts.len() {
        let num: i32 = parts[i].parse().unwrap();
        let mut next_num = parts[0].parse().unwrap();

        if i != parts.len() - 1 {
            next_num = parts[i + 1].parse().unwrap();
        }

        if num == next_num {
            sumr += num;
        }
    }

    return sumr;
}

fn part_2(parts: Vec<&str>) -> i32 {
    let mut sumr = 0;
    let halfway: usize = parts.len() / 2;

    for i in 0..parts.len() {
        let num: i32 = parts[i].parse().unwrap();
        let mut next_pos: usize = i + halfway;

        if next_pos >= parts.len() {
            next_pos = next_pos - parts.len();
        }

        let next_num: i32 = parts[next_pos].parse().unwrap();

        if num == next_num {
            sumr += num;
        }
    }

    return sumr;
}
